Source/FakeItEasy.IntegrationTests/DummyTests.cs

```diff
@@ -38,6 +39,33 @@ public void Correct_exception_should_be_thrown_when_dummy_is_requested_for_non_f
             }
         }
 
+        [SuppressMessage("Microsoft.Naming", "CA2204:Literals should be spelled correctly", MessageId = "CanCreateDummyOfType",
+            Justification = "That's really the name of the method.")]
+        [Test]
+        public void Definition_for_type_should_be_cached_after_first_use()
+        {
+            var firstDummy = A.Dummy<Dummy>();
+
+            var secondDummy = A.Dummy<Dummy>();
+
+            Assert.That(secondDummy.Count, Is.EqualTo(firstDummy.Count), "CanCreateDummyOfType was called the wrong number of times.");

```

Comment:
Ah, I see. I'm not keen in the resetting either so I think your original suggestion was best.

Perhaps we can trim the names down a bit? E.g.:

```csharp
[Test]
public void Definition_for_type_should_be_cached_after_first_use()
{
    A.Dummy<Dummy>();
    var firstCallCount = DummyTestsDummyDefinition.CanCreateDummyOfTypeCallCount;

    A.Dummy<Dummy>();
    var secondCallCount = DummyTestsDummyDefinition.CanCreateDummyOfTypeCallCount;

    firstCallCount.Should().Be(
        secondCallCount, "once a dummy has been created for a type, its dummy definition should be cached");
}
```